platform/darwin/src/MGLCircle.h

```diff
@@ -0,0 +1,95 @@
+#import <Foundation/Foundation.h>
+#import <CoreLocation/CoreLocation.h>
+
+#import "MGLShape.h"
+#import "MGLGeometry.h"
+
+#import "MGLTypes.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ An `MGLCircle` object represents a closed, circular shape also known as a
+ <a href="https://en.wikipedia.org/wiki/Spherical_cap">spherical cap</a>. A
+ circle is defined by a center coordinate, specified as a
+ `CLLocationCoordinate2D` instance, and a physical radius measured in meters.
+ The circle is approximated as a polygon with a large number of vertices and
+ edges. Due to the map’s Spherical Mercator projection, a large enough circle
+ appears as an elliptical or even sinusoidal shape. You could use a circle to
+ visualize for instance an impact zone, the `CLLocation.horizontalAccuracy` of a
+ GPS location update, the regulated airspace around an airport, or the
+ addressable consumer market around a city.
+
+ You can add a circle overlay directly to a map view using the
```

Comment:
Do you think it is necessary to call out that this type does not support all the same paint/layout options as a shape added to an `MGLFillStyleLayer`?  I'm thinking specifically of fill patterns, but there are probably others as well.